src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlExceptionActionProcessor.cs

```diff
@@ -33,9 +35,13 @@ public Task Execute(TRequest request, TException exception, CancellationToken ca
                 {
                     throw new SqlQueryPlanException(Core.Resources.SqlQueryProcessorRanOutOfInternalResourcesException);
                 }
+                else if (sqlException.Number == LoginFailedForUser)
+                {
+                    throw new LoginFailedForUserException(Core.Resources.LoginFailedForUser);
```

Comment:
Should this message just be the default of this exception?

Reply:
For now I think this makes more sense. There isn't much a FHIR Client can do about an error with the database connection which is between the app and sql